src/Monocle/Api/Server.hs

```diff
@@ -905,16 +917,37 @@ handleLogin uriM = do
     loc <- O.prepareAuthenticationRequestUrl (mkSessionStore oidcenv Nothing uriM) oidc [O.openId] mempty
     return (show loc)
 
+-- This session cookie is made for the Monocle Web APP to figure out the authenticated user context
+makeMonocleCookie :: LBS.ByteString -> SetCookie
+makeMonocleCookie jwt = do
+  defaultSetCookie
+    { setCookieValue = LBS.toStrict jwt
+    , setCookieHttpOnly = False -- This make the cookie readable from the Web APP
```

Comment:
Is this set so that the username can be read by the webUI from the auth cookie? If yes, then we should keep the http-only, and use an extra query to get the info from a API response body.

Reply:
Yes, this cookie is only for the Web APP to be able to read the authenticated user info. However this is **NOT** the auth cookie. This cookie is the "Monocle" cookie which contain an **invalidated** JWT containing the same information as the one in the local storage previously.

The auth cookie is the JWT-Cookie created by the acceptLogin function of servant-auth-servant. This one is httpOnly.